call-home/src/bin/callhome/collector/k8s_client.rs

```diff
@@ -17,15 +22,33 @@ impl K8sClient {
 
     /// Get number of nodes present in the cluster.
     pub(crate) async fn get_node_len(&self) -> Result<usize, K8sResourceError> {
+        let mut nodes_count: usize = 0;
+
         let nodes: Api<Node> = Api::all(self.client.clone());
-        let list = nodes.list(&Default::default()).await?;
-        Ok(list.items.len())
+        let mut list_params = ListParams::default().limit(KUBE_API_PAGE_SIZE);
+
+        loop {
+            let node_list = nodes.list_metadata(&list_params).await?;
+
+            let continue_ = node_list.metadata.continue_.clone();
+
+            nodes_count += node_list.items.len();
+
+            match continue_ {
+                Some(token) => {
+                    list_params = list_params.continue_token(token.as_str());
```

Comment:
Seems pointless to clone only to use reference later on

```suggestion
            nodes_count += node_list.items.len();       

            match node_list.metadata.continue_.as_str() {
                Some(token) => {
                    list_params = list_params.continue_token(token);
```